Gratia Trouble (SOFTWARE-4442) (#775)
opensciencegrid/docs: docs/other/troubleshooting-gratia.md

Gratia/GRACC: The Big Picture
-----------------------

> Gratia is software used in OSG to gather accounting information.

Reviewer: ```suggestion
Gratia is software used in OSG to account for usage of computational resources.
```

Author: I actually prefer the original phrasing.  (It's definitely not something i would change in terms of correctness.)

If you want to say something about "computational resources" specifically, maybe we could say "to gather accounting information for usage of computational resources."

Also I haven't decided how i feel about describing "storage probes" as collecting usage info about "computational resources."

Reviewer: > "to gather accounting information for usage of computational resources."

Fine by me.

> Also I haven't decided how i feel about describing "storage probes" as collecting usage info about "computational resources."

I don't think we have to worry about the storage probe case.

Author: ok

> The information is collected from individual resources at a site, such as a Compute Entrypoint or a a submission host.

Reviewer: ```suggestion
The information is collected from individual resources at a site, such as a Compute Entrypoint or a submission host.
```

The program that collects the data is called a "Gratia probe".

> The information is transferred to the central OSG GRACC server.

Reviewer: ```suggestion
The information is transferred to the [central OSG GRACC server](https://gracc.opensciencegrid.org/).
```

Author: I don't think this link would fit here.

- This link already appears later under "You can see the OSG's GRACC website at \[URL\]".
- As seen in the diagram right below this paragraph, this document makes a distinction between the "GRACC server" and the "GRACC website".  The text hear discusses the "server" but you are giving a link to the "website".

So i think adding this link is redundant and only serves to confuse things.

Reviewer: Sounds good to me.

| SSLHost             | The GRACC server this probe reports to                                                |
| SSLRegistrationHost | The GRACC server this probe reports to                                                |
| ProbeName           | The unique name for this probe. Note that it includes the probe type and the host name |

> | SiteName            | The name of your Resource, as registered in [OSG Topology](/common/registration).      |

Reviewer: ```suggestion
| SiteName            | The name of your Resource, as registered in [OSG Topology](../common/registration.md).      |
```

New MkDocs version, new link format requirements: https://opensciencegrid.org/technology/documentation/style-guide/#links

Author: >MkDocs 1.0.0 does not support site-relative links (e.g. /software/development-process.md). You must use document-relative links (e.g. ../software/development-process.md) instead.

What a slap in the face, MkDocs.

!!! note 

>     Condor rotates history files, so you can only report what Condor has kept.

Reviewer: ```suggestion
    HTCondor rotates history files, so you can only report what Condor has kept.
```

Author: I dunno, there are a lot of references to "Condor" (the batch system) throughout the document, including two in this sentence.  Do you want to convert them all?

Reviewer: Yeah, let's strive to use the official name for software in our docs. I would imagine that non-Emacs editors can handle a find-and-replace :)

Author: Yes although it's slightly more nuanced because sometimes Condor refers to HTCondor and sometimes Condor refers to the condor probe (ie, gratia-probe-condor).

Reviewer: Ah, that's very confusing! In the latter case, let's make it clearer with something like `the HTCondor Gratia probe` or `gratia-probe-condor`

Author: Alrighty

>     Controlling the Condor history is documented in the Condor manual.

Reviewer: ```suggestion
    Controlling the HTCondor history is documented in the HTCondor manual.
```

    In particular, see the options for
    [MAX_HISTORY_LOG](https://htcondor.readthedocs.io/en/latest/admin-manual/configuration-macros.html#MAX_HISTORY_ROTATIONS)
    and

>     [MAX_HISTORY_ROTATIONS](https://htcondor.readthedocs.io/en/latest/admin-manual/configuration-macros.html#MAX_HISTORY_ROTATIONS).

Reviewer: This looks good but I'm not sure if this section is all that relevant anymore since we package the appropriate config by default:

- https://github.com/opensciencegrid/gratia-probe/blob/master/condor/99_gratia.conf
- https://github.com/opensciencegrid/gratia-probe/blob/master/rpm/gratia-probe.spec#L469

Reviewer: This entire subsection (https://github.com/opensciencegrid/docs/blob/9106411e5fa17f067ee6122534b383144994b4b3/docs/other/troubleshooting-gratia.md#if-you-accidentally-did-not-set-per_job_history_dir-see-above) that this note lives in.

Author: Ok.  Not sure if it might still be relevant for non-rpm installs?  Anyway, dropping the section.

Reviewer: > Not sure if it might still be relevant for non-rpm installs?

The thought of this for Gratia probe frightens me!
